src/datamodel/CIFAR10DataModel.py

```diff
@@ -0,0 +1,39 @@
+from torchvision import datasets, transforms
+from torch.utils.data import random_split, DataLoader
+import pytorch_lightning as pl
+from typing import Optional
+
+class CIFAR10DataModule(pl.LightningDataModule):
+    def __init__(self, data_dir: str = "./data"):
+        super().__init__()
+        self.data_dir = data_dir
+        self.transform = transforms.ToTensor()
+
+    def prepare_data(self):
+        # download
+        datasets.CIFAR10('data', train=True, download=True, transform=self.transform)
+
+    def setup(self, stage: Optional[str] = None):
+        # Assign train/val datasets for use in dataloaders
+        if stage == "fit" or stage is None:
+            cifar10_full = datasets.CIFAR10(self.data_dir, train=True, transform=self.transform)
+            self.train, self.val = random_split(cifar10_full, [45000, 5000])
+
+        # Assign test dataset for use in dataloader(s)
+        if stage == "test" or stage is None:
+            self.test = datasets.CIFAR10(self.data_dir, train=False, transform=self.transform)
+
+        if stage == "predict" or stage is None:
+            self.predict = datasets.CIFAR10(self.data_dir, train=False, transform=self.transform)
+
+    def train_dataloader(self):
```

Comment:
You are mostly right for the data module case, but what if I want to pass in other arguments like num_workers, drop_last, shuffle, sampler?

Reply:
I tried to solve the problem by using dictionary variables, but I don't know whether it works well.